leshan-server-core/src/main/java/org/eclipse/leshan/server/observation/ObservationListener.java

```diff
@@ -30,5 +30,13 @@
      * @param observation the observation for which new data are received
      * @param reponse the lwm2m response received
      */
-    void newValue(Observation observation, ObserveResponse response);
+    void onResponse(Observation observation, ObserveResponse response);
```

Comment:
I like the newly introduced error handler method and I also think that we should rename `newValue()` :-)
My understanding is that `onResponse` will be invoked with ANY response received from the LWM2M client that we are able to process whereas `onError` is invoked when we cannot process the response, right? In particular this means that responses sent form the client containing error codes (4.xx) will also be delivered via the `onResponse` method if the content can be processed correctly, right?